app/assets/javascripts/discourse/app/helpers/raw.js

```diff
@@ -5,16 +5,21 @@ import { RUNTIME_OPTIONS } from "discourse-common/lib/raw-handlebars-helpers";
 import { buildResolver } from "discourse-common/resolver";
 
 let resolver;
+let viewsByTemplateName = new Map();
```

Comment:
When caching stuff in a module local variable like this, we also need to clear it between tests.

Generally the pattern is to export a `resetCache()` or similar method and make sure that is called in test runs.